android/src/main/kotlin/com/abilia/system_settings_editor/SystemSettingsEditorPlugin.kt

```diff
@@ -8,28 +15,116 @@ import io.flutter.plugin.common.MethodChannel
 import io.flutter.plugin.common.MethodChannel.MethodCallHandler
 import io.flutter.plugin.common.MethodChannel.Result
 
-/** SystemSettingsEditorPlugin */
-class SystemSettingsEditorPlugin: FlutterPlugin, MethodCallHandler {
-  /// The MethodChannel that will the communication between Flutter and native Android
-  ///
-  /// This local reference serves to register the plugin with the Flutter Engine and unregister it
-  /// when the Flutter Engine is detached from the Activity
+import android.provider.Settings
+import android.util.Log
+import android.view.WindowManager
+import androidx.annotation.RequiresApi
+import androidx.core.content.ContextCompat.startActivity
+import io.flutter.embedding.engine.plugins.activity.ActivityAware
+import io.flutter.embedding.engine.plugins.activity.ActivityPluginBinding
+
+class SystemSettingsEditorPlugin: FlutterPlugin, MethodCallHandler, ActivityAware {
   private lateinit var channel : MethodChannel
+  private lateinit var context : Context
+  private lateinit var activity : Activity
 
   override fun onAttachedToEngine(@NonNull flutterPluginBinding: FlutterPlugin.FlutterPluginBinding) {
     channel = MethodChannel(flutterPluginBinding.binaryMessenger, "system_settings_editor")
     channel.setMethodCallHandler(this)
+    context = flutterPluginBinding.applicationContext
+  }
+
+  override fun onAttachedToActivity(binding: ActivityPluginBinding) {
+    this.activity = binding.activity;
   }
 
+  @RequiresApi(Build.VERSION_CODES.M)
   override fun onMethodCall(@NonNull call: MethodCall, @NonNull result: Result) {
-    if (call.method == "getPlatformVersion") {
-      result.success("Android ${android.os.Build.VERSION.RELEASE}")
+    if (!Settings.System.canWrite(context)) {
+      openAndroidPermissionsMenu()
+      result.error("ACCESS", "Cannot write to system settings. Permission needed.", null)
     } else {
-      result.notImplemented()
+      when (call.method) {
+        "getBrightness" -> result.success(getBrightness())
+        "setBrightness" -> setBrightnessHandler(call, result)
+        else -> result.notImplemented()
+      }
+    }
+  }
+
+  @RequiresApi(Build.VERSION_CODES.M)
+  private fun openAndroidPermissionsMenu() {
+    val intent = Intent(Settings.ACTION_MANAGE_WRITE_SETTINGS)
+    intent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK)
+    intent.data = Uri.parse("package:" + activity.packageName)
+    startActivity(context, intent, null)
+  }
+
+  private fun getBrightness(): Double {
+    val cResolver: ContentResolver = context.contentResolver
+    try {
+      Settings.System.putInt(cResolver,
+        Settings.System.SCREEN_BRIGHTNESS_MODE, Settings.System.SCREEN_BRIGHTNESS_MODE_MANUAL)
+      val brightness = Settings.System.getInt(cResolver, Settings.System.SCREEN_BRIGHTNESS)
+      val max = getBrightnessMax()
+      return brightness.toDouble() / max
```

Comment:
Testing on an emulator, and setting the screen brightness to 55% in Android settings, and in the example app it says 13.7%

Not sure what causes this inconsistency